lua/resession/init.lua

```diff
@@ -69,11 +81,47 @@ M.get_current = function()
   return tab_sessions[tabpage] or current_session
 end
 
----Detach from the current session
-M.detach = function()
-  current_session = nil
-  local tabpage = vim.api.nvim_get_current_tabpage()
-  tab_sessions[tabpage] = nil
+---Detach from a session
+---@param name? string
+---@param opts? resession.DetachOpts
+---@param target_tabpage? integer
```

Comment:
Both `name` and `target_tabpage` are optional, so it makes more sense to put them in the `opts`. Might also want to check & error if both `name` and `target_tabpage` are provided. A final small naming nit: I don't think that "target" helps clarify anything here; we can probably just simplify it to `tabpage`.

Reply:
Yeah, I makes sense, I initially called it `tabpage` since I saw it used as a parameter name elsewhere.